### PR TITLE
fix(document templates): #28 add CSAF document template service

### PR DESCRIPTION
**Pull request by rainer-exxcellent**
Extend Rest service to provide prefilled CSAF documents templates

### CODE REVIEW

File: src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package de.bsi.secvisogram.csaf_cms_backend.model.template;
```
Comment (valentindey) [inline src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateServiceTest.java:1]: There's a discrepancy in naming. The files in the main package all are called `DocumentTemplate...` whereas in the tests the `Document...` part is skipped. I plead in favor of consistency to remove the `Document` prefix altogether.

**Review by valentindey — CHANGES_REQUESTED** [review]
Github shows several warnings about files missing 

image

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 2]
```diff
@@ -105,7 +110,7 @@ public ResponseEntity<AdvisoryResponse> readCsafDocument(
         LOG.info("readCsafDocument");
         checkValidUuid(advisoryId);
         try {
-            return ResponseEntity.ok(advisoryService.getAdvisory(advisoryId));
+            return (ResponseEntity.ok(advisoryService.getAdvisory(advisoryId)));
```
[the remark this answers is not in the document]
Comment (rainer-exxcellent) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:113]: ok

File: src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.junit.jupiter.api.Test;
+
+@SuppressFBWarnings(value = "CLI_CONSTANT_LIST_INDEX", justification = "for test it is ok")
+public class TemplateTest {
+
+    @Test
+    public void expressionToJson() throws JsonProcessingException {
```
[the remark this answers is not in the document]
Comment (rainer-exxcellent) [reply src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateTest.java:14]: ok

File: src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateServiceTest.java (hunk 1 above)
Comment (rainer-exxcellent) [reply src/test/java/de/bsi/secvisogram/csaf_cms_backend/model/template/TemplateServiceTest.java:1]: Ok

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 4]
```diff
@@ -312,17 +322,27 @@ public List<AdvisoryTemplateInformationResponse> listAllTemplates() {
...
             @PathVariable
             @Parameter(
                     in = ParameterIn.PATH,
                     description = "The ID of the template to read."
-            ) long templateId
+            ) String templateId
     ) {
 
         // only for debugging, remove when implemented
         LOG.info("readTemplate {}", sanitize(templateId));
```
[the remark this answers is not in the document]
Comment (rainer-exxcellent) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:334]: That logging i very helpful while testing in development. 
We should keep them during development

**Review by valentindey — APPROVED** [review]

**Pull request merged**